Error with new exports field on package.json

### Describe the bug
i recently created a fork of shoelace and while testing it out, we are seeing a error that does not happens in production release but happens on new versions built... Looks like it is related to this [PR](https://github.com/shoelace-style/shoelace/pull/1007)

If i remove the new exports block, it work fine... i'm missing anything?

```
ERROR in ./src/externals/vendors.js 83:0-66
Module not found: Error: Package path ./dist/themes/light.css is not exported from package C:\Programacao\platform-cortex\modules\competitiva\frontend\node_modules\@cortex-intelligence\design-system (see exports field in C:\Programacao\platform-cortex\modules\competitiva\frontend\node_modules\@cortex-intelligence\design-system\package.json)

ERROR in ./src/externals/vendors.js 84:0-77
Module not found: Error: Package path ./dist/components/button/button.js is not exported from package C:\Programacao\platform-cortex\modules\competitiva\frontend\node_modules\@cortex-intelligence\design-system (see exports field in C:\Programacao\platform-cortex\modules\competitiva\frontend\node_modules\@cortex-intelligence\design-system\package.json)

ERROR in ./src/externals/vendors.js 85:0-73
Module not found: Error: Package path ./dist/components/icon/icon.js is not exported from package C:\Programacao\platform-cortex\modules\competitiva\frontend\node_modules\@cortex-intelligence\design-system (see exports field in C:\Programacao\platform-cortex\modules\competitiva\frontend\node_modules\@cortex-intelligence\design-system\package.json)

ERROR in ./src/externals/vendors.js 87:0-93
Module not found: Error: Package path ./dist/utilities/base-path.js is not exported from package C:\Programacao\platform-cortex\modules\competitiva\frontend\node_modules\@cortex-intelligence\design-system (see exports field in C:\Programacao\platform-cortex\modules\competitiva\frontend\node_modules\@cortex-intelligence\design-system\package.json)
```

### To Reproduce
I trying to import a fresh fork of this project to a existing project, like this:
```
import '@cortex-intelligence/design-system/dist/themes/light.css';
import '@cortex-intelligence/design-system/dist/components/button/button.js';
import '@cortex-intelligence/design-system/dist/components/icon/icon.js';

import { setBasePath } from '@cortex-intelligence/design-system/dist/utilities/base-path.js';
// Set the base path to the folder you copied Shoelace's assets to
setBasePath('/client/js/design-system');
```

### Demo

Tell me if its required, if so i create it.

### Screenshots
If applicable, add screenshots to help explain the bug.

### Browser / OS
 - OS: Windows 11
 - Browser: Chrome
 - Browser version: Latest

### Additional information

@tao-cumplido @samuelstroschein I tested some variations:

```json
"./dist/components/*/*.js": "./dist/components/*/*.js",
```

```bash
ERROR in ./src/externals/vendors.js 85:0-73
Module not found: Error: Package path ./dist/components/icon/icon.js is not exported from package C:\Programacao\platform-cortex\modules\competitiva\frontend\node_modules\@cortex-intelligence\design-system (see exports field in C:\Programacao\platform-cortex\modules\competitiva\frontend\node_modules\@cortex-intelligence\design-system\package.json)
```


```json
"./dist/components/*.js": "./dist/components/*.js",
```

```bash
ERROR in ./src/externals/vendors.js 85:0-73
Module not found: Error: Package path ./dist/components/icon/icon.js is not exported from package C:\Programacao\platform-cortex\modules\competitiva\frontend\node_modules\@cortex-intelligence\design-system (see exports field in C:\Programacao\platform-cortex\modules\competitiva\frontend\node_modules\@cortex-intelligence\design-system\package.json)
```


The only way that seens to work is (or without dist folder, but would require all the users to change their imports, is that wanted?):
```json
"exports": {
    ".": {
      "types": "./dist/shoelace.d.ts",
      "import": "./dist/shoelace.js"
    },
    "./dist/themes/*": "./dist/themes/*",
    "./dist/components/*": "./dist/components/*",
    "./dist/utilities/*": "./dist/utilities/*",
    "./dist/react/*": "./dist/react/*",
    "./dist/translations/*": "./dist/translations/*"
  },
```

I included the utilities, that is needed to `setBasePath`, but there's also others folders in the module, like `react`, `translations`, `styles`, `internal` ... is any of these needed? `react` and `translations` looks like yes...

--- a/package.json
+++ b/package.json
@@ -14,7 +14,6 @@
       "types": "./dist/shoelace.d.ts",
       "import": "./dist/shoelace.js"
     },
-    "./shoelace.js": "./dist/shoelace.js",
     "./themes/*": "./dist/themes/*",
     "./components/*": "./dist/components/*",
     "./utilities/*": "./dist/utilities/*",

--- a/package.json
+++ b/package.json
@@ -13,7 +13,13 @@
     ".": {
       "types": "./dist/shoelace.d.ts",
       "import": "./dist/shoelace.js"
-    }
+    },
+    "./shoelace.js": "./dist/shoelace.js",
+    "./themes/*": "./dist/themes/*",
+    "./components/*": "./dist/components/*",
+    "./utilities/*": "./dist/utilities/*",
+    "./react/*": "./dist/react/*",
+    "./translations/*": "./dist/translations/*"
   },
   "files": [
     "dist"

